src/clojure_lsp/crawler.clj

```diff
@@ -167,12 +168,56 @@
     (concat (diagnose-unused-aliases uri declared-aliases unused-aliases)
             (diagnose-unused-references uri declared-references all-envs))))
 
-(defn find-diagnostics [project-aliases uri usages]
-  (let [unknown (diagnose-unknown project-aliases usages)
+(defn- kondo-finding->diagnostic [lines {:keys [message level row col] :as finding}]
+  (let [decrow (max 0 (dec row))
+        deccol (max 0 (dec col))
+        ^String line (when (> (count lines) decrow)
+                       (nth lines decrow))
+        start-char (when (> (count line) deccol)
+                     (.charAt line deccol))
+        expression? (identical? \( start-char)
+        finding (merge {:end-row row :end-col col} finding)]
+    {:range (shared/->range (if expression?
+                              (assoc finding :end-row row :end-col col)
+                              finding))
+    :message (str (string/upper-case (str (first message))) (string/join (rest message)))
+    :severity (case level
+                :error 1
+                :warning 2
+                :info 3)
+    :source "clj-kondo"}))
+
+(def kondo-base-args {:cache true
+                      :cache-dir ".clj-kondo/cache"
+                      :config {:linters (into {} (map #(hash-map % {:level :off})
+                                                      #{:invalid-arity
+                                                        :unused-bindings
+                                                        :unresolved-symbol
+                                                        :unresolved-namespace
+                                                        :unused-namespace}))}})
+
+(defn- run-kondo!
+  ([paths]
+   (kondo/run! (merge kondo-base-args {:lint paths})))
+  ([text lang]
+   (with-in-str text (kondo/run! (merge kondo-base-args {:lint ["-"]
+                                                         :lang lang})))))
+
+(defn- kondo-find-diagnostics [uri text]
+  (let [file-type (shared/uri->file-type uri)
+        {:keys [findings]} (run-kondo! text file-type)
+        lines (string/split-lines text)]
+    (->> findings
+         (filter #(= "<stdin>" (:filename %)))
+         (map (partial kondo-finding->diagnostic lines)))))
+
+(defn find-diagnostics [project-aliases uri text usages]
+  (let [kondo-chan (async/go (kondo-find-diagnostics uri text))
+        unknown (diagnose-unknown project-aliases usages)
         unused (diagnose-unused uri usages)
         unknown-forwards (diagnose-unknown-forward-declarations usages)
         wrong-arity (diagnose-wrong-arity uri usages)
-        result (concat unknown unused unknown-forwards wrong-arity)]
+        result (concat unknown unused unknown-forwards (async/<!! kondo-chan) wrong-arity)]
```

Comment:
I think the idea is to in the future replace the clojure-lsp lint with clj-kondo only, so we can start with a feature flag that toggle kondo lint or clojure-lsp lint, what do you think?
Something like [:use-clj-kondo?](https://github.com/snoe/clojure-lsp/blob/master/src/clojure_lsp/refactor/transform.clj#L298) on the settings would be a good approach

Reply:
> This is concating clj-kondo lints with clojure-lsp lints right?

Yep.

> I think the idea is to in the future replace the clojure-lsp lint with clj-kondo only, so we can start with a feature flag that toggle kondo lint or clojure-lsp lint, what do you think?

I also have the idea that we can switch over to clj-kondo, but I kind of made this PR as a starting point and gradually we can replace clojure-lsp's linting functionalities with those currently disabled kondo linters, then finally we can consider replacing the lsp's parser with kondo's analysis output. What are your thoughts this route?

Reply:
I see, so this PR adds kondo to lint things that clojure-lsp doesn't yet, right?
After this PR, we can start replacing each diagnose like `unknown`, `unused`, `wrong-arity` and others with clj-kondo, right?
I'm Ok with that, I think this way we don't need the feature flag that much

Reply:
Yes, that's exactly what I meant :) Hopefully we can collect some feedback as the switch progresses.

Reply:
Perfect!
Do you know some lint that returns from this new integration?
I could only reproduce at my machine the linters from clojure-lsp

Reply:
Try this bizarre form:

```clojure
(let []
  (do
    {1}))
```

kondo produces three messages on my machine. Also here's a handy list of all linters in kondo: https://github.com/borkdude/clj-kondo/blob/master/src/clj_kondo/impl/config.clj#L13

Reply:
Nice, it really works nice, in the future maybe append to the message the linter used for that diagnostic like clojure-lsp does (to help the user find which linter is that if he pretends to disable it)